include/dlaf/init.h

```diff
@@ -48,6 +50,12 @@ struct Init {
   static void finalize() {}
 };
 
+template <>
+void Init<Backend::MC>::initialize(configuration const&);
+template <>
+void Init<Backend::MC>::finalize();
+extern template struct Init<Backend::MC>;
+
```

Comment:
I forgot to add a comment, so I  will do it here: 

LGTM overall. 

The triangular solver has to be adapted to the new MPI API. The implementation of the MPI executor allows to be compatible with `getMPIExecutor<>()` API.

Reply:
> Do you need this in the header for testing purposes?

Good question, I don't remember (it wasn't for testing purposes as far as I can remember). It'd indeed be nice if it can go in the cpp file. Will check again.

> I forgot to add a comment, so I will do it here:
> 
> LGTM overall.
> 
> The triangular solver has to be adapted to the new MPI API. The implementation of the MPI executor allows to be compatible with `getMPIExecutor<>()` API.

Yeah, I'm changing that right now. Given that the executor's constructor now chooses the correct pool `getMPIExecutor` is somewhat unnecessary. It might be worth keeping it just for symmetry with the other executors though.

Did I understand it correctly that one doesn't actually need a separate row and column executor (they're exactly the same)? I presume this was from trying to have the executor know which type it is (which we should still look at)?

Reply:
> Did I understand it correctly that one doesn't actually need a separate row and column executor (they're exactly the same)? I presume this was from trying to have the executor know which type it is (which we should still look at)?

Yes, the row/column distinction comes from the communicator which is now handled outside of the executor. The only exception is if we want to later bring back "blocking" MPI version of the executor. In that case we would need to save the index of the OS thread to which the executor is assigned. That means we would need two different executors for the row and column communicators. However, as we decided to remove the "blocking" version, a single executor is sufficient.

Reply:
> Do you need this in the header for testing purposes?

It was unnecessary, I've moved it to the cpp file.

For the triangular solver I now just use one executor for both column and row communication. It should be easy enough to change it back when/if needed.

Reply:
I agree with you about having a single Executor for both row and col...

I have one more question: why not having a single Executor also for all MPI communications? I mean, for the moment is not a big deal because it does not take any parameter, but it may be reasonable to have it inside our `initialize` and get it via the `getMPIExecutor()`. Do you see any problem? This may be the topic of a dedicated issue/PR anyway...

Reply:
My wording was maybe confusing. The same executor is used for both types of communication now. If you're talking about using literally the same instance that's also a possibility (i.e. returning a reference to a function static executor); currently it's a new instance every time you call `getMPIExecutor`, but I don't see that as a huge overhead (executors are cheap to create).